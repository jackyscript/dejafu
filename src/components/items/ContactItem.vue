<script setup>
defineProps({
  id: {
    type: String,
    required: true
  },
  value: {
    type: String,
    required: false
  }
})
</script>
<template>
  <section>
    <slot name="icon"></slot>
    <h3><slot name="heading"></slot></h3>
    <p v-if="value !== undefined" />
    <slot name="link">{{ value }}</slot>
  </section>
</template>

<style scoped>

h3 {
  line-height: 1.2;
	padding: 0;
	background: var(--pico-mark-gradient);
	background-clip: text;
}
</style>
