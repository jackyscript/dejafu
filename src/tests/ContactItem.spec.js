import { mount } from '@vue/test-utils'
import ContactItem from '@/components/items/ContactItem.vue'
import { expect, describe, it } from 'vitest'

describe('ContactItem.vue', () => {
  it('can be mounted', () => {
    expect(ContactItem).toBeTruthy()
  })

  it("renders the contact's value", () => {
    const wrapper = mount(ContactItem, {
      props: {
        id: 'contact',
        value: 'Jacky Fu'
      }
    })

    expect(wrapper.text()).toContain('Jacky Fu')
  })

  it('matches the snapshot', () => {
    const wrapper = mount(ContactItem, {
      props: {
        id: 'contact',
        value: 'Dezhi Fu'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
