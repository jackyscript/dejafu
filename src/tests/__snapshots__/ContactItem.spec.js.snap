// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactItem.vue > matches the snapshot 1`] = `
"<section data-v-f2e26872="">
  <article data-v-f2e26872="">
    <h3 data-v-f2e26872=""></h3>
    <p data-v-f2e26872=""></p>Dezhi Fu
  </article>
</section>"
`;
